django_rest_api/user/serializers.py

```diff
@@ -1,20 +1,42 @@
-from django import forms
-from django.contrib.auth.models import User
 from rest_framework import serializers
 from django.contrib.auth import authenticate
-from rest_framework.validators import UniqueValidator
-from rest_framework.authtoken.models import Token
-from .models import CustomUser  
+from .models import CustomUser
 
 
+class RegisterSerializer(serializers.ModelSerializer):
+    password1 = serializers.CharField(
+        write_only=True, 
+        required=True
+    )
+    password2 = serializers.CharField(
+        write_only=True, 
+        required=True
+    )
+    
+    class Meta:
+        model = CustomUser
+        fields = ['username', 'email', 'password1', 'password2']
+
+    def validate(self, data):
+        if data['password1'] != data['password2']:
+            raise serializers.ValidationError({"password": "Passwords must match."})
+        return data
+    
+    def create(self, validated_data):
+        return CustomUser.objects.create_user(
+            username=validated_data['username'],
+            email=validated_data['email'],
+            password=validated_data['password1']
+        )
+
 
 class LoginSerializer(serializers.Serializer):
     username = serializers.CharField(required=True)
     password = serializers.CharField(write_only=True, required=True)
     
-    # class Meta:
-    #     model = CustomUser
-    #     fields = ['username', 'password']
+    class Meta:
```

Comment:
No need for Meta class if you're using regular `Serializer` and not `ModelSerializer`.